x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js

```diff
@@ -0,0 +1,24 @@
+(function propertiesbuilder(attr) {
+    var props = {};
+
+    props["user"] = attr["username"];
+    props["password"] = attr["password"];
+
+    var extraProps = attr[connectionHelper.attributeWarehouse];
+    if (extraProps != null && extraProps.trim().length > 0) {
+        var avps = extraProps.trim().split(/[\s&]/);
+        for (var i = 0; i < avps.length; i++) {
+            var tokens = avps[i].split("=");
+            if (tokens.length != 2 || tokens[0].length == 0 || tokens[1].length == 0) {
+                logging.log("Malformed AVP: [" + avps[i] + "] in additional params: [" + extraProps + "]");
```

Comment:
I guess what AVP means, but I am not sure. Especially this being a logged message, I'd give it a meaningful name. Something like "Malformed attribute-value pair:......"

Reply:
Acronym expanded.